libs/parallelism/algorithms/include/hpx/parallel/algorithms/transform_inclusive_scan.hpp

```diff
@@ -86,53 +484,57 @@ namespace hpx { namespace parallel { inline namespace v1 {
         }
 
         ///////////////////////////////////////////////////////////////////////
-        template <typename FwdIter2>
+        template <typename IterPair>
         struct transform_inclusive_scan
-          : public detail::algorithm<transform_inclusive_scan<FwdIter2>,
-                FwdIter2>
+          : public detail::algorithm<transform_inclusive_scan<IterPair>,
+                IterPair>
         {
             transform_inclusive_scan()
               : transform_inclusive_scan::algorithm("transform_inclusive_scan")
             {
             }
 
-            template <typename ExPolicy, typename InIter, typename OutIter,
-                typename Conv, typename T, typename Op>
-            static OutIter sequential(ExPolicy, InIter first, InIter last,
-                OutIter dest, Conv&& conv, T&& init, Op&& op)
+            template <typename ExPolicy, typename InIter, typename Sent,
+                typename OutIter, typename Conv, typename T, typename Op>
+            static constexpr util::in_out_result<InIter, OutIter> sequential(
+                ExPolicy, InIter first, Sent last, OutIter dest, Conv&& conv,
+                T&& init, Op&& op)
             {
                 return sequential_transform_inclusive_scan(first, last, dest,
```

Comment:
I'm surprised this compiles. `sequential_transform_inclusive_scan` returns a single iterator that shouldn't be convertible to an `in_out_result`

Reply:
I take this back, sorry for the noise.